File: frontend/apps/web-antd/src/api/profiles.ts
import { del, patch, post } from './request';
import type { Profile } from './dashboard';

export interface ProfileSaveBody {
  id?: string;
  name?: string;
  authMode?: string;
  region?: string;
  tenantId?: string;
  clientId?: string;
  clientSecret?: string;
  driveId?: string;
  siteId?: string;
  siteHostname?: string;
  sitePath?: string;
  libraryName?: string;
  rootPath?: string;
  capacityEnabled?: boolean;
  capacityPoolId?: string;
}

export const profilesApi = {
  save: (body: ProfileSaveBody) => post<{ ok: true; profile: Profile }>('/api/profiles', body),
  setCapacityEnabled: (id: string, capacityEnabled: boolean) =>
    patch<{ ok: true; profile: Profile }>(`/api/profiles/${id}/capacity`, { capacityEnabled }),
  setCapacityPool: (id: string, capacityPoolId: string) =>
    patch<{ ok: true; profile: Profile }>(`/api/profiles/${id}/capacity-pool`, { capacityPoolId }),
  remove: (id: string) => del(`/api/profiles/${id}`)
};

File: backend/schemas/profiles.py
from __future__ import annotations

from .common import FlexibleSchema


class ProfileUpsertRequest(FlexibleSchema):
    id: str | None = None
    name: str | None = None
    authMode: str | None = None
    region: str | None = None
    tenantId: str | None = None
    clientId: str | None = None
    clientSecret: str | None = None
    refreshToken: str | None = None
    driveId: str | None = None
    siteId: str | None = None
    siteHostname: str | None = None
    sitePath: str | None = None
    libraryName: str | None = None
    rootPath: str | None = None
    capacityPoolId: str | None = None


class ProfileResponse(FlexibleSchema):
    profile: dict


class ProfileCapacityRequest(FlexibleSchema):
    capacityEnabled: bool


class ProfileCapacityPoolRequest(FlexibleSchema):
    capacityPoolId: str
